packages/xo-web/src/xo-app/proxies/index.js

```diff
@@ -114,21 +109,116 @@ const COLUMNS = [
     itemRenderer: proxy => <Vm id={proxy.vmUuid} link />,
     name: _('vm'),
   },
+  {
+    itemRenderer: (proxy, { upgradesByProxy, upgradeAppliance }) => {
+      const globalState = upgradesByProxy[proxy.id]
+      if (globalState === undefined) {
+        return
+      }
+
+      const { state } = globalState
+      if (state.endsWith('-upgrade-needed')) {
+        return (
+          <div>
+            <ActionButton
+              btnStyle='success'
+              disabled={proxy.vmUuid === undefined}
+              handler={upgradeAppliance}
+              handlerParam={proxy.id}
+              icon='upgrade'
+            >
+              {_('upgrade')}
+            </ActionButton>
+            <p className='text-warning'>
+              <Icon icon='alarm' />
+              &nbsp;{_('upgradesAvailable')}
+            </p>
+          </div>
+        )
+      }
+
+      if (
+        state === 'xoa-up-to-date' ||
+        state === 'xoa-upgraded' ||
+        state === 'updater-upgraded' ||
+        state === 'installer-upgraded'
+      ) {
+        return (
+          <ActionButton
+            btnStyle='primary'
```

Comment:
Why this button style?

IMHO, it's too visible, `primary` should be used to indicate the main and important action on an object, it's not the case here as the proxy is already up-to-data.